app/javascript/packages/document-capture/higher-order/with-background-encrypted-upload.jsx

```diff
@@ -1,8 +1,49 @@
 import React, { useContext } from 'react';
 import UploadContext from '../context/upload';
 
+/**
+ * Returns a promise resolving to an DataView representation of the given Blob object.
+ *
+ * @param {Blob} blob Blob object.
+ *
+ * @return {Promise<DataView>}
+ */
+export function blobToDataView(blob) {
+  return new Promise((resolve, reject) => {
+    const reader = new window.FileReader();
+    reader.onload = ({ target }) => {
+      resolve(new DataView(/** @type {ArrayBuffer} */ (target?.result)));
+    };
+    reader.onerror = () => reject();
```

Comment:
this is going to discard whatever error message we got from the reader --- do we want that? Otherwise we could just do `reader.onerror = reject;` right?

Reply:
> this is going to discard whatever error message we got from the reader --- do we want that? Otherwise we could just do `reader.onerror = reject;` right?

Correct, it would discard. It was something where in adapting `blob.arrayBuffer` to this `FileReader` approach, it wasn't clear if or when a rejection would actually occur, but because `onerror` exists on `FileReader`, that we may as well wire it to `reject`.

`onerror` passes a message, not an error object, but AFAIK either can be valid for `reject`, so your suggestion _should_ work fine. I'll update :+1:

Reply:
> `onerror` passes a message, not an error object, but AFAIK either can be valid for `reject`, so your suggestion _should_ work fine. I'll update 👍

I must have been reading the wrong documentation. It passes an _event_ object. The error is available on the instance of the FileReader.

Documentation: https://developer.mozilla.org/en-US/docs/Web/API/FileReader/onerror

Updated (with test case) in 0a28828.